### PR TITLE
Convert to dual language: python 2 and 3

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

File: docs/doc_tools/all_link_anchors_to_lowercase.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 def list_all_html_files():
     html_files = list()
     for root, dirs, files in os.walk("../"):
-        # html_files.extend(filter(lambda x: x.endswith(".html"), files))
+        # html_files.extend(filter(lambda x: x. endswith(".html"), files))
```
Comment (LegacyKing) [inline docs/doc_tools/all_link_anchors_to_lowercase.py]: I know Python, and x.endswith should not contain a space, that would not follow convention.

**Comment by grimreaper** [comment]
fixed

File: docs/doc_tools/all_link_anchors_to_lowercase.py (hunk 1 above)
Comment (LegacyKing) [inline docs/doc_tools/all_link_anchors_to_lowercase.py]: Not sure what you "Fixed", the same change is still present.

**Comment by grimreaper** [comment]
re-resolved. Some of my git magic overwrote the fix

**Pull request merged**